### PR TITLE
add class decorator

### PR DESCRIPTION
**Pull request by ijosue16**
In this PR, I did the following:
 
 1. Create a `Nameable` to raise `NotImplementedError` if the `correct_name` method is not implemented.
 2. Create a `Decorator` base class
 3. Create a `CapitalizeDecorator` class to capitalize the name
 4. Create a `TrimmerDecorator` class to trim long name

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require_relative 'nameable'
+class Decorator
```
Comment (zieeco) [inline decorator.rb]: - Hey, Your decorator class should inherit from the Nameable class. Please. kindly fix this.

Comment (zieeco) [reply decorator.rb]: - Hey, regarding the decorator should inherit from the Nameable class, please, also kindly add this missing correct_name method it inherits from the Nameable class.

```ruby
def correct_name
      @nameable.correct_name
    end
```

File: person.rb [hunk 2]
```diff
@@ -20,3 +27,10 @@ def can_use_services?
     of_age? || parent_permission
   end
 end
+
+person = Person.new(22, 'maximilianus')
+person.correct_name
+capitalizedperson = CapitalizeDecorator.new(person)
+capitalizedperson.correct_name
+capitalizedtrimmedperson = TrimmerDecorator.new(capitalizedperson)
+capitalizedtrimmedperson.correct_name
```
Comment (zieeco) [inline person.rb:36]: - Hey, after making the changes suggested, your code should run properly.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @ijosue16 ,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Fair class decoration ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @ijosue16,

✔️ Class decorator is well implemented

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**